app/services/forest_liana/ability/permission/smart_action_approval.rb

```diff
@@ -0,0 +1,67 @@
+require 'jwt'
+
+module ForestLiana
+  module Ability
+    module Permission
+    class SmartActionApproval
+
+      def initialize(parameters, collection, smart_action, user)
+        @parameters = parameters
+        @collection = collection
+        @smart_action = smart_action
+        @user = user
+      end
+
+      def can_trigger?
+        if @smart_action['triggerEnabled'].include?(@user['roleId']) && @smart_action['approvalRequired'].exclude?(@user['roleId'])
+          return true if @smart_action['triggerConditions'].empty? || match_conditions('triggerConditions')
+
+        elsif @parameters[:data][:attributes][:signed_approval_request].present? && @smart_action['userApprovalEnabled'].include?(@user['roleId'])
+          deserialize
+          if ((@smart_action['userApprovalConditions'].empty? || match_conditions('userApprovalConditions')) &&
+            (@parameters[:data][:attributes][:requester_id] != @user['id'] || @smart_action['selfApprovalEnabled'].include?(@user['roleId']))
+          )
+            return true
+          end
+        elsif @smart_action['approvalRequired'].include?(@user['roleId'])
+          if @smart_action['approvalRequiredConditions'].empty? || match_conditions('approvalRequiredConditions')
+            raise ForestLiana::Ability::Exceptions::RequireApproval.new(@smart_action['userApprovalEnabled'])
+          end
```

Comment:
I think you need to put this condition first. As an approver can also have the right to trigger but in case of an approval (`signed_approval_request` param present) we want to check approval rights.

```
if @parameters[:data][:attributes][:signed_approval_request].present? && @smart_action['userApprovalEnabled'].include?(@user['roleId'])
```

Reply:
🤔 to discuss